nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/StreamWorkManager.java

```diff
@@ -141,6 +148,8 @@ public void schedule(Work work, Scheduling scheduling, boolean afterCommit) {
         }
         String key = work.getPartitionKey();
         appender.append(key, Record.of(key, WorkComputation.serialize(work)));
+        KeyValueStore kvStore = Framework.getService(KeyValueService.class).getKeyValueStore(KV_NAME);
+        kvStore.put(work.getId() + STATE, Work.State.SCHEDULED.toString(), 60 * 60L);
```

Comment:
Use a constant like
STATE_TIMEOUT_S = 3600L;
but I am afraid we will need to have a configurable value, 
after all, we are trying to handle long running work which is something subjective.